source/common/http/conn_manager_impl.cc

```diff
@@ -1743,9 +1766,11 @@ Tracing::OperationName ConnectionManagerImpl::ActiveStream::operationName() cons
   return connection_manager_.config_.tracingConfig()->operation_name_;
 }
 
-const std::vector<Http::LowerCaseString>&
-ConnectionManagerImpl::ActiveStream::requestHeadersForTags() const {
-  return connection_manager_.config_.tracingConfig()->request_headers_for_tags_;
+const std::vector<Tracing::CustomTag>& ConnectionManagerImpl::ActiveStream::customTags() const {
+  if (!tracing_custom_tags_.empty()) {
+    return tracing_custom_tags_;
+  }
+  return connection_manager_.config_.tracingConfig()->custom_tags_;
```

Comment:
dont we always merge this into the `tracing_custom_tags_` vector? can we just always return `tracing_custom_tags_`?

Reply:
it's ok to always merge, fixed